src/components/Support/Support.jsx

```diff
@@ -61,9 +62,10 @@ function formatMoney(number) {
   return str;
 }
 
-export default class Support extends React.Component {
+export default class Support extends VisibilitySensor {
```

Comment:
2 things:
1. And most important one, this break the goal of this project, @skipjack vision when rebuilding this project is to focus solely on  **content**, meaning that custom plugins/components/logic/code etc should be outside of it, as dependencies, that's why my custom MDX plugins are dependencies.2

2. This way of writing React components is not recommended by React itself (https://reactjs.org/docs/composition-vs-inheritance.html#so-what-about-inheritance), meaning that it is not idiomatic, which is important if you want to attract new contributors, especially in an Open Source project like this one.

Reply:
this looks like an overkill to me, @skipjack thoughts?

Reply:
I agree with @montogeek in this case.

> ...when rebuilding this project is to focus solely on content, meaning that custom plugins/components/logic/code etc should be outside of it, as dependencies, that's why my custom MDX plugins are dependencies.

Yeah it seems there are already some pretty popular open source packages that do similar things so we should consider those first instead of bloating this codebase with more components:

- https://github.com/joshwnj/react-visibility-sensor
- https://github.com/fkhadra/react-on-screen
- https://github.com/dazld/react-on-visible

Even if none of those work, you could always write this new component as a separate package the same way we've done for other things.

> This way of writing React components is not recommended by React itself (https://reactjs.org/docs/composition-vs-inheritance.html#so-what-about-inheritance), meaning that it is not idiomatic, which is important if you want to attract new contributors, especially in an Open Source project like this one.

Yeah I agree with this as well. However, if you use one of the open source packages I listed above then the new component (and inheritance) wouldn't even be necessary.

Reply:
I've started this PR with a `react-visibility-sensor` which doesn't benefit from Intersection Observer. After looking at our user base i was tempted to go ahead and use that. Which seems to be widely supported nowadays.
So i had did a re-work here within PR.

having that said i will move the visibility sensor into a separate package and rework it from inheritance type module into a HOC

Reply:
Thanks for dropping by btw @skipjack 👍